2.0.1: Accessing myexternalip.com can fail.

**Short Summary:**
When accessing myexternalip.com to get our public ip, the web site can be unavailable, which causes cluster creation to fail.

**Expected Results:**
When we need to find out what our public ip is, we need to possibly attempt this from multiple reliable sources. @taaraora  suggested:
```
ips, err := net.LookupIP("google.com")
```

**Actual Results:**
If the website is down or unavailable, we get the below error when spinning up a cluster
```
DEBU[2018-11-29T22:39:45+02:00] error while accessing http://myexternalip.com/raw
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x14cc2d1]

goroutine 479 [running]:
github.com/supergiant/control/pkg/workflows/steps/amazon.findOutBoundIP(0x0, 0x0, 0x0, 0x0)
        /c/go-path/src/github.com/supergiant/control/pkg/workflows/steps/amazon/util.go:50 +0x71
github.com/supergiant/control/pkg/workflows/steps/amazon.FindOutboundIP(0x1c57b80, 0xc001f80780, 0x1ae9508, 0x0, 0x0, 0x0, 0xc001f80780)
        /c/go-path/src/github.com/supergiant/control/pkg/workflows/steps/amazon/util.go:26 +0x183
github.com/supergiant/control/pkg/workflows/steps/amazon.(*CreateSecurityGroupsStep).whiteListSupergiantIP(0xc00000e4c8, 0x1c57b80, 0xc001f80780, 0x1c8fb20, 0xc000eac0c0, 0xc002f4a540, 0x14, 0xc0024eee40, 0x17)
        /c/go-path/src/github.com/supergiant/control/pkg/workflows/steps/amazon/create_security_groups.go:223 +0x5c
github.com/supergiant/control/pkg/workflows/steps/amazon.(*CreateSecurityGroupsStep).Run(0xc00000e4c8, 0x1c57b80, 0xc001f80780, 0x1c3ae60, 0xc0000c2008, 0xc000889800, 0x0, 0x0)
        /c/go-path/src/github.com/supergiant/control/pkg/workflows/steps/amazon/create_security_groups.go:145 +0x645
github.com/supergiant/control/pkg/provisioner.(*TaskProvisioner).preProvision(0xc0000c6ac0, 0x1c57b80, 0xc001f80780, 0xc000889800, 0x4, 0x1c35a10)
        /c/go-path/src/github.com/supergiant/control/pkg/provisioner/provisioner.go:286 +0xee
github.com/supergiant/control/pkg/provisioner.(*TaskProvisioner).ProvisionCluster.func1(0xc0000c6ac0, 0x1c57b80, 0xc001f80780, 0xc000889800, 0xc0000e16d0, 0xc0003ae168, 0x1, 0x1, 0xc002afcc00, 0x3, ...)
        /c/go-path/src/github.com/supergiant/control/pkg/provisioner/provisioner.go:116 +0x6a
created by github.com/supergiant/control/pkg/provisioner.(*TaskProvisioner).ProvisionCluster
        /c/go-path/src/github.com/supergiant/control/pkg/provisioner/provisioner.go:115 +0x458
exit status 2
```

--- a/pkg/workflows/steps/amazon/util.go
+++ b/pkg/workflows/steps/amazon/util.go
@@ -41,8 +41,8 @@ func FindOutboundIP(ctx context.Context, findExternalIP func() (string, error)) 
 
 func findOutBoundIP() (string, error) {
 	serviceURLs := []string{
-		"http://myexternalip.com/raw",
 		"http://checkip.amazonaws.com/",
+		"http://myexternalip.com/raw",
 	}
 
 	var (

--- a/pkg/workflows/steps/amazon/util.go
+++ b/pkg/workflows/steps/amazon/util.go
@@ -50,8 +50,11 @@ func findOutBoundIP() (string, error) {
 		err error
 	)
 
+	client := &http.Client{
+		Timeout: time.Second * 10,
+	}
 	for _, serviceURL := range serviceURLs {
-		resp, err := http.Get(serviceURL)
+		resp, err := client.Get(serviceURL)
 		if err != nil {
 			logrus.Debugf("error while accessing %s", serviceURL)
 			continue

--- a/pkg/workflows/steps/amazon/util.go
+++ b/pkg/workflows/steps/amazon/util.go
@@ -40,22 +40,38 @@ func FindOutboundIP(ctx context.Context, findExternalIP func() (string, error)) 
 }
 
 func findOutBoundIP() (string, error) {
-	serviceURL := "http://myexternalip.com/raw"
-
-	resp, err := http.Get(serviceURL)
-	if err != nil {
-		logrus.Debugf("error while accessing %s", serviceURL)
-		return "", err
+	serviceURLs := []string{
+		"http://myexternalip.com/raw",
+		"http://checkip.amazonaws.com/",
 	}
 
-	defer resp.Body.Close()
-	res, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return "", err
-	}
+	var (
+		publicIP string
+		err error
+	)
 
-	publicIP := string(res)
-	publicIP = strings.Replace(publicIP, "\n", "", -1)
+	for _, serviceURL := range serviceURLs {
+		resp, err := http.Get(serviceURL)
+		if err != nil {
+			logrus.Debugf("error while accessing %s", serviceURL)
+			continue
+		}
+
+		res, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			continue
+		}
+
+		publicIP = string(res)
+		publicIP = strings.Replace(publicIP, "\n", "", -1)
+
+		logrus.Debugf("get IP from %s", serviceURL)
+		if resp != nil && resp.Body != nil {
+			resp.Body.Close()
+		}
+
+		break
+	}
 
 	return publicIP, err
 }

--- a/pkg/workflows/steps/amazon/util.go
+++ b/pkg/workflows/steps/amazon/util.go
@@ -51,6 +51,7 @@ func findOutBoundIP() (string, error) {
 	)
 
 	client := &http.Client{
+		Transport: http.DefaultTransport,
 		Timeout: time.Second * 10,
 	}
 	for _, serviceURL := range serviceURLs {

